File: frontend/src/api/employee.ts
import type { PagedResult } from './product'
import { get, post, put } from './request'

/** 在职状态（0 离职 / 1 在职） */
export type EmployeeStatus = 0 | 1

/** 性别（0 未填 / 1 男 / 2 女） */
export type Gender = 0 | 1 | 2

/** 员工列表行（对应后端 EmployeeListItemDto） */
export interface Employee {
  id: string
  employeeNo: string
  name: string
  gender: Gender | null
  phone: string | null
  departmentId: string | null
  departmentName: string | null
  positionId: string | null
  positionName: string | null
  hireDate: string
  resignDate: string | null
  status: EmployeeStatus
  statusText: string
  userId: string | null
  userDisplayName: string | null
  createdAt: string
}

/** 员工详情（对应后端 EmployeeDetailDto） */
export interface EmployeeDetail extends Employee {
  email: string | null
  remark: string | null
  updatedAt: string
}

/** 员工可选账号（对应后端 EmployeePickUserDto） */
export interface EmployeePickUser {
  id: string
  username: string
  displayName: string
}

export interface GetEmployeesParams {
  page?: number
  pageSize?: number
  keyword?: string
  departmentId?: string
  positionId?: string
  status?: EmployeeStatus
}

export interface CreateEmployeePayload {
  employeeNo: string
  name: string
  gender?: Gender
  phone?: string
  email?: string
  departmentId?: string
  positionId?: string
  hireDate: string
  resignDate?: string
  status: EmployeeStatus
  userId?: string
  remark?: string
}

/** 编辑载荷：不含工号（创建后不可改，AGENTS.md §4.5） */
export interface UpdateEmployeePayload {
  name: string
  gender?: Gender
  phone?: string
  email?: string
  departmentId?: string
  positionId?: string
  hireDate: string
  resignDate?: string
  status: EmployeeStatus
  userId?: string
  remark?: string
}

/** 分页查询员工 */
export function getEmployees(params: GetEmployeesParams): Promise<PagedResult<Employee>> {
  return get<PagedResult<Employee>>('/employees', { params })
}

/** 查询员工详情 */
export function getEmployee(id: string): Promise<EmployeeDetail> {
  return get<EmployeeDetail>(`/employees/${id}`)
}

/** 新增员工 */
export function createEmployee(payload: CreateEmployeePayload): Promise<EmployeeDetail> {
  return post<EmployeeDetail>('/employees', payload)
}

/** 编辑员工（工号不可改） */
export function updateEmployee(id: string, payload: UpdateEmployeePayload): Promise<EmployeeDetail> {
  return put<EmployeeDetail>(`/employees/${id}`, payload)
}

/** 在职 / 离职切换 */
export function updateEmployeeStatus(id: string, status: EmployeeStatus): Promise<EmployeeDetail> {
  return put<EmployeeDetail>(`/employees/${id}/status`, { status })
}

/** 可选账号来源：启用且未被绑定的账号 ∪ 当前员工已绑定的账号 */
export function getAvailableUsers(employeeId?: string): Promise<EmployeePickUser[]> {
  return get<EmployeePickUser[]>('/employees/available-users', {
    params: employeeId ? { employeeId } : undefined,
  })
}

File: backend/src/App.Core/Entities/Employee.cs
namespace App.Core.Entities;

/// <summary>
/// 员工实体（对应 PostgreSQL 表 Employees）。
/// **员工 ≠ 账号**：员工经可空、唯一的 <see cref="UserId"/> 绑定一个系统账号（登录一律走 009 的 Users）；
/// 员工不做删除，仅「在职 ↔ 离职」切换，保留历史与审计引用。
/// </summary>
public sealed class Employee
{
    /// <summary>员工 ID</summary>
    public Guid Id { get; set; }

    /// <summary>工号，全局唯一，创建后不可修改</summary>
    public string EmployeeNo { get; set; } = string.Empty;

    /// <summary>姓名</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>性别，可空</summary>
    public Gender? Gender { get; set; }

    /// <summary>手机号，可空；非空时全局唯一</summary>
    public string? Phone { get; set; }

    /// <summary>邮箱，可空；非空时全局唯一</summary>
    public string? Email { get; set; }

    /// <summary>所属部门 id，可空</summary>
    public Guid? DepartmentId { get; set; }

    /// <summary>所属岗位 id，可空</summary>
    public Guid? PositionId { get; set; }

    /// <summary>入职日期（纯日期，无时区语义）</summary>
    public DateOnly HireDate { get; set; }

    /// <summary>离职日期，可空；置离职时为空则由 Handler 补当天</summary>
    public DateOnly? ResignDate { get; set; }

    /// <summary>在职状态（在职 / 离职）</summary>
    public EmployeeStatus Status { get; set; } = EmployeeStatus.Active;

    /// <summary>关联系统账号 id，可空；非空时全局唯一（一个账号最多被一个员工绑定）</summary>
    public Guid? UserId { get; set; }

    /// <summary>备注，可空</summary>
    public string? Remark { get; set; }

    /// <summary>创建时间</summary>
    public DateTimeOffset CreatedAt { get; set; }

    /// <summary>更新时间</summary>
    public DateTimeOffset UpdatedAt { get; set; }

    /// <summary>创建人用户 id</summary>
    public Guid? CreatedBy { get; set; }

    /// <summary>更新人用户 id</summary>
    public Guid? UpdatedBy { get; set; }
}
